test/cmocka/src/math/arithmetic/exponential.c

```diff
@@ -0,0 +1,91 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/*
+ * Copyright(c) 2022 Intel Corporation. All rights reserved.
+ *
+ * Author: Shriram Shastry <malladi.sastry@linux.intel.com>
+ */
+
+#include <stdio.h>
+#include <stdint.h>
+#include <stdarg.h>
+#include <stddef.h>
+#include <setjmp.h>
+#include <math.h>
+#include <cmocka.h>
+#include <string.h>
+#include <time.h>
+#include <stdlib.h>
+
+#include <sof/math/exp_fcn.h>
+#include <sof/audio/format.h>
+#include <rtos/string.h>
+#include <sof/common.h>
+
+#define ULP_TOLERANCE 5.60032793
+#define ULP_SCALE 0.0000999999999749
+
+static void gen_exp_testvector(double a, double b, double *r, int32_t *b_i);
+
+/* testvector in Q4.28, -5 to +5 */
+/*
+ * Arguments	:double a
+ *		 double b
+ *		 double *r
+ *		 int32_t *b_i
+ * Return Type	:void
+ */
+static void gen_exp_testvector(double a, double b, double *r, int32_t *b_i)
+{
+	double a2;
+	double b2;
+
+	*r = a + rand() % (int32_t)(b - a + 1);
+	a2 = *r * 268435456;
+	b2 = fabs(a2);
+	if (b2 < 4503599627370496)
+		a2 = floor(a2 + 0.5);
+
+	if (a2 < 2147483648)
+		*b_i = (a2 >= -2147483648) ? a2 : INT32_MIN;
```

Comment:
Quickbuild is back - with a vengeance. This fails to compile with xt-xcc in quickbuild, see error below.

Welcome to integer promotion rules in C!

The error is unfortunately visible only on the internal Intel server since we had to migrate the public 01.org to new hosting. The internal build number is build/11377029

```
09:17:28,736 INFO  - cd /quickbuild/workspace1/24733/SOF_FW/build_ut/test/cmocka/src/math/arithmetic && /localdisk/tools/xtensa/RG-2017.8-linux/XtensaTools/bin/xt-xcc -DBLD_COUNTERS=0 -DCONFIG_NUMBERS_NORM -DCONFIG_NUMBERS_VECTOR_FIND -DUNIT_TEST -D_UINTPTR_T_DEFINED -DRELATIVE_FILE=\"test/cmocka/src/math/arithmetic/base_e_logarithm.c\" -I/quickbuild/workspace1/24733/SOF_FW/test/cmocka/include -I/localdisk/tools/cmocka/build_cavs2x_LX6HiFi3_2017_8/install/include -I/quickbuild/workspace1/24733/SOF_FW/src/arch/xtensa/include -I/quickbuild/workspace1/24733/SOF_FW/src/arch/xtensa/xtos -I/quickbuild/workspace1/24733/SOF_FW/xtos/include -I/quickbuild/workspace1/24733/SOF_FW/src/platform/tigerlake/include -I/localdisk/tools/xtensa/RG-2017.8-linux/cavs2x_LX6HiFi3_2017_8/xtensa-elf/arch/include -I/localdisk/tools/xtensa/RG-2017.8-linux/cavs2x_LX6HiFi3_2017_8/xtensa-elf/include -I/quickbuild/workspace1/24733/SOF_FW/src/platform/intel/cavs/include -I/quickbuild/workspace1/24733/SOF_FW/src/arch/xtos-wrapper/include -I/quickbuild/workspac...
09:17:28,815 INFO  - cc1: warnings being treated as errors
09:17:28,815 INFO  - cmocka/src/math/arithmetic/exponential.c: In function ‘gen_exp_testvector’:
09:17:28,815 INFO  - cmocka/src/math/arithmetic/exponential.c:45: warning: integer constant is too large for ‘long’ type
09:17:28,815 INFO  - cmocka/src/math/arithmetic/exponential.c:48: warning: this decimal constant is unsigned only in ISO C90
09:17:28,815 INFO  - cmocka/src/math/arithmetic/exponential.c:49: warning: this decimal constant is unsigned only in ISO C90
09:17:28,818 INFO  - cmocka/src/math/arithmetic/CMakeFiles/exponential.dir/build.make:76: recipe for target 'test/cmocka/src/math/arithmetic/CMakeFiles/exponential.dir/exponential.c.o' failed
```

Reply:
should be corrected!
Can you confirm, please?
Thanks